tests/cross_functional/system_test/test_mcg_bucket_notification.py

```diff
@@ -0,0 +1,300 @@
+        to guarantee the correct teardown order. Otherwise the pvc factory teardown
+        might fail when deleting a PVC that the BucketNotificationsManager is still using.
+
+        Returns:
+            BucketNotificationsManager: An instance of the BucketNotificationsManager class
+
+        """
+        notif_manager = BucketNotificationsManager()
+        notif_manager.pvc_factory = pvc_factory_class
+        request.addfinalizer(notif_manager.cleanup)
+
+        notif_manager.setup_kafka()
+        return notif_manager
+
+    def verify_events(
+        self, notify_manager, topic, bucket_names, event_name, timeout=120, sleep=5
+    ):
+
+        for events in TimeoutSampler(
+            timeout=timeout,
+            sleep=sleep,
+            func=notify_manager.get_events,
+            topic=topic,
+        ):
+            for event in events:
+                if (
+                    event["eventName"] == event_name
+                    and event["s3"]["bucket"]["name"] in bucket_names
+                ):
+                    logger.info(
+                        f'{event_name} event found for the bucket {event["s3"]["bucket"]["name"]}'
+                    )
+                    bucket_names.remove(event["s3"]["bucket"]["name"])
+
+            if len(bucket_names) == 0:
+                logger.info(f"Verified {event_name} for all the buckets.")
+                break
+
+    @polarion_id("OCS-6406")
+    def test_bucket_notification_system_test(
+        self,
+        nodes,
+        notify_manager,
+        bucket_factory,
+        reduce_expiration_interval,
+        awscli_pod,
+        test_directory_setup,
+        mcg_obj,
+        noobaa_db_backup_and_recovery,
+        snapshot_factory,
+        setup_mcg_bg_features,
+        validate_mcg_bg_features,
+    ):
+        """
+        This is to test MCG bucket notification feature on multiple MCG buckets
+        with bucket notification enabled for multiple s3 operation. We also perform some
+        Noobaa specific disruptive operations to verify notification is not affected
+
+        Steps:
+        1. Enable bucket notifications on the NooBaa CR
+        2. Create a Kafka topic and add a Kafka notification connection to the NooBaa CR
+        3. Create 5 buckets and configure bucket notification for each of the s3 operation
+        4. Perform Object upload and verify object upload events are received for all the buckets
+        5. Tag object from the bucket and restart noobaa pod nodes.
+           Verify ObjectTagging:Put events has occurred for all the buckets.
+        6. Remove object from the bucket and restart kafka pods. Verify ObjectRemoved:Delete event has occurred
+        7. Enable object expiration for 1 day and expire the objects manually.
+           Perform Noobaa db backup and recovery.
+           Make sure notifications are updated for LifecycleExpiration:Delete event
+
+        """
+
+        NUM_OF_BUCKETS = 5
```

Comment:
how did you derive this number? can it be increased

Reply:
Me and @hnallurv discussed and came with the number. generally customer cases have about 5-10 buckets. It can be increased